integrationTests/vm/esdt/esdtNFTSCs_test.go

```diff
@@ -1 +1,331 @@
 package esdt
+
+import (
+	"encoding/hex"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/integrationTests"
+	"github.com/stretchr/testify/require"
+)
+
+func TestESDTNFTIssueCreateBurnSendViaAsyncViaExecuteOnSC(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+	advertiser, nodes, idxProposers := createNodesAndPrepareBalances(1)
+
+	defer func() {
+		_ = advertiser.Close()
+		for _, n := range nodes {
+			_ = n.Messenger.Close()
+		}
+	}()
+
+	initialVal := big.NewInt(10000000000)
+	integrationTests.MintAllNodes(nodes, initialVal)
+
+	round := uint64(0)
+	nonce := uint64(0)
+	round = integrationTests.IncrementAndPrintRound(round)
+	nonce++
+
+	scAddress := deployNonPayableSmartContract(t, nodes, idxProposers, &nonce, &round)
+
+	issuePrice := big.NewInt(1000)
+	txData := []byte("nftIssue" + "@" + hex.EncodeToString([]byte("TOKEN")) +
+		"@" + hex.EncodeToString([]byte("TKR")))
+	integrationTests.CreateAndSendTransaction(
+		nodes[0],
+		nodes,
+		issuePrice,
+		scAddress,
+		string(txData),
+		integrationTests.AdditionalGasLimit+core.MinMetaTxExtraGasCost,
+	)
+
+	time.Sleep(time.Second)
+	nrRoundsToPropagateMultiShard := 10
+	nonce, round = integrationTests.WaitOperationToBeDone(t, nodes, nrRoundsToPropagateMultiShard, nonce, round, idxProposers)
+	time.Sleep(time.Second)
+
+	tokenIdentifier := string(getTokenIdentifier(nodes))
+	txData = []byte("setLocalRoles" + "@" + hex.EncodeToString(scAddress) +
+		"@" + hex.EncodeToString([]byte(tokenIdentifier)) + "@" + "03" + "@" + "05")
+	integrationTests.CreateAndSendTransaction(
+		nodes[0],
+		nodes,
+		big.NewInt(0),
+		scAddress,
+		string(txData),
+		integrationTests.AdditionalGasLimit+core.MinMetaTxExtraGasCost,
+	)
+
+	time.Sleep(time.Second)
+	nonce, round = integrationTests.WaitOperationToBeDone(t, nodes, nrRoundsToPropagateMultiShard, nonce, round, idxProposers)
+	time.Sleep(time.Second)
+
+	txData = []byte("nftCreate" + "@" + hex.EncodeToString([]byte(tokenIdentifier)) +
```

Comment:
a job for tx data builder :D

Reply:
I've got used to old style tx creation :dagger: